pkg/baremetal/v1alpha3/register.go

```diff
@@ -0,0 +1,46 @@
+package v1alpha3
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+// +k8s:deepcopy-gen=false
+type BareMetalProviderSpecCodec struct {
+	encoder runtime.Encoder
+	decoder runtime.Decoder
+}
+
+const GroupName = "cluster.weave.works"
+
+var SchemeGroupVersion = schema.GroupVersion{Group: GroupName, Version: "v1alpha3"}
+
+var (
+	SchemeBuilder      runtime.SchemeBuilder
+	localSchemeBuilder = &SchemeBuilder
+	AddToScheme        = localSchemeBuilder.AddToScheme
+)
+
+func init() {
+	localSchemeBuilder.Register(addKnownTypes)
+}
+
+func addKnownTypes(scheme *runtime.Scheme) error {
+	scheme.AddKnownTypes(SchemeGroupVersion,
+		&BareMetalMachine{},
+		&BareMetalMachineList{},
+		&BareMetalCluster{},
+		&BareMetalClusterList{},
+	)
+	metav1.AddToGroupVersion(scheme, SchemeGroupVersion)
+	return nil
+}
+
+func NewScheme() (*runtime.Scheme, error) {
```

Comment:
This function seems unused?

Reply:
![image](https://user-images.githubusercontent.com/8125524/85547762-1ec53d80-b616-11ea-8075-a1b948d6e71d.png)